### ISSUE TITLE
Can't insert rows in a closed section: 0

### ISSUE THREAD

**Issue by victor-sarda** [fuzz-productions/FZAccordionTableView#29]
Hi, 
I got this error and I can't figure out why ? 
Can you explain me what does it means ? I followed your example in my code.

Thank you

**Comment by kgaidis** [comment]
I would have to see the code for your particular case, however, there might actually be a "bug" in this case.

Think of inserting rows as a two step process:

1. Modify **your** data source (some array) by inserting an element.
2. Tell the table view that there has been a modification by calling `insertRowsAtIndexPaths`. We can't see `UITableView` internal code, but we can assume that this then modifies some internal data structure in a table view to tell it that there is a new row. IF that row is visible, then table view animates the insertion.

In the case of `FZAccordionTableView`, IF a section is closed, no rows are visible, which means that the `UITableView` thinks that the row count for the section is 0. Inserting a row by calling `insertRowsAtIndexPaths` WHILE a section is "closed" doesn't make sense because the section is closed, there is no reason to animate a row insertion if the section is closed. 

In other words:
-  a "closed" section represents a row count of 0.
- inserting an row in a closed section means a row count of at least 1, which is a contradiction. 

You could try just modifying your data structure (probably an array), and skip the `insertRowsAtIndexPaths` part. Then hopefully, when the user taps the section you inserted a row for, everything will work smoothly. However, I am not 100% sure if this is true. There is a chance that right now you simply can't add rows to a closed section, unless you call `tableView.reloadData()`, and not `insertRowsAtIndexPaths`. This is something that might warrant a fix. It's an edge case, but something that might be easily handled.

**Comment by victor-sarda** [comment]
Hi,
Thank you for answering.
It was my data structure indeed I was doing it wrong. Your answer helped me,
**Thanks a lot**
